feature/gribi/ate_tests/hierarchical_weight_resolution_test/hierarchical_weight_resolution_test.go

```diff
@@ -270,40 +287,50 @@ func (a *attributes) configInterfaceDUT(t *testing.T, d *ondatra.DUTDevice, p *o
 		i = a.NewOCInterface(p.Name())
 	}
 
-	a.configSubinterfaceDUT(t, i)
-	intfPath := gnmi.OC().Interface(p.Name())
-	gnmi.Update(t, d, intfPath.Config(), i)
 	if *deviations.ExplicitPortSpeed {
-		fptest.SetPortSpeed(t, p)
+		i.GetOrCreateEthernet().PortSpeed = fptest.GetIfSpeed(t, p)
 	}
+
+	a.configSubinterfaceDUT(t, i)
```

Comment:
can u please use the following code 
```
	a.configSubinterfaceDUT(t, i)
	intfPath := gnmi.OC().Interface(p.Name())
	a.configL3SubifDUT(t, i)
	gnmi.Replace(t, d, intfPath.Config(), i)
	fptest.LogQuery(t, "DUT", intfPath.Config(), gnmi.GetConfig(t, d, intfPath.Config()))
	a.assignSubifsToNetworkInstance(t, d, p)
	
```

Reply:
Could you elaborate on this?

Previously I've been told that IP addresses have to be configured after interfaces are assigned to VRFs in some implementations.. Is that not the case anymore?

Reply:
there is deviation InterfaceConfigVrfBeforeAddress but that is not used by us.

Reply:
Okay, that's good.
So, just to settle it once and for all,
1) you're ok with fully-configured subinterface before the vrf assignment, right?
2) do you need an address (or full subinterface config) before the assignment? I.e. the currently proposed scheme will not work for you? Or it doesn't matter

(sorry for so many questions; it's just we've been changing it back and forth while trying to accommodate all vendor requirements, so want to get maximum clarity)

Reply:
@mojiiba any feedback on the above two questions?

as an additional point, in the upstream main version of the test, currently the same order is used (L3 addressing is configured after VRF assignment). So I'm not entirely sure why are you asking to change this now in this PR (unless it also breaks something for you)

Reply:
@mojiiba can you help give us some hint? Thanks.

Reply:
currently, we do not allow gnmi replace on the assigned interfaces to non-default vrfs. so, either you need to configure interfaces and then assign them to non-default vrf or use gnmi.update instead of replace.  so to answer your question we are ok with option 1.

Reply:
> so, either you need to configure interfaces and then assign them to non-default vrf or use gnmi.update instead of replace.

but the code was fully aligned with that?
the order of operations was:
1) gnmi.Replace (to create a subif)
2) assign
3) gnmi.Update (to add l3 info)

anyhow, switched to another method based on conversation with @xw-g.